base/common/python/pki/nssdb.py

```diff
@@ -440,37 +444,59 @@ def convert_db(self):
 
         logger.info("Migration successful")
 
-    def module_exists(self, name):
+    def module_exists(self, name, library=None):
 
-        logger.info('Checking module %s', name)
+        def regex_in_output(cmd, pattern):
+            result = self.run(cmd, capture_output=True, check=True)
+            output = result.stdout.decode('utf-8')
 
-        cmd = [
-            'modutil',
-            '-dbdir', self.directory,
-            '-rawlist'
-        ]
+            p = re.compile(re.escape(pattern))
+            logger.info('Looking for: %s', pattern)
 
-        result = self.run(cmd, capture_output=True, check=True)
-        output = result.stdout.decode('utf-8')
+            for line in output.splitlines():
+                line = line.strip()
 
-        pattern = re.compile(r' name="%s"' % name)
+                if not line:
+                    continue
 
-        for line in output.splitlines():
-            line = line.strip()
+                logger.info('Output: %s', line)
 
-            if not line:
-                continue
+                if p.search(line):
+                    return True
 
-            logger.info('Output: %s', line)
+            return False
 
-            if pattern.search(line):
+        logger.info('Checking module %s', name)
+
+        commands = [
+            [
+                'modutil',
+                '-dbdir', self.directory,
+                '-rawlist'
+            ],
+            [
+                'p11-kit',
+                'list-modules'
+            ],
+        ]
+
+        patterns = [
+            r'name="%s"' % name,
+            r'%s: %s' % (name, library if library else '')
+        ]
+
+        for i in range(len(commands)):
+            if regex_in_output(commands[i], patterns[i]):
```

Comment:
This approach is confusing.  There is a 1:1 match between command and pattern, so a list of pairs might be easier to understand:

```python
pairs = [
  ( ['modutil', '-dbdir', self.directory, '-rawlist'], r'name="%s"' % escaped_name ),
  ( ['p11-kit', 'list-modules'], r'%s: %s' % (escaped_name, escaped_library if library else '') ),
]

for (command, pattern) in pairs:
  if regex_in_output(command, pattern):
    ...
```

Reply:
Sure I can combine it, it found it less readable.

Reply:
Thanks.  I just think indexing into two different arrays was a bit confusing, and perhaps more prone to mistakes during maintenance.